ads/google/utils.js

```diff
@@ -133,3 +134,31 @@ function validateDimensions(width, height, widthCond, heightCond, errorBuilder)
   }
   return !badParams.length;
 }
+
+/** @const {Number} */
+const WHITELISTED_HEIGHT = 150;
+
+/**
+ * Check that the <amp-ad> tag height is one of the allowed heights for
+ * responsive ad units.
+ * @param {Number} height
+ */
+export function validateResponsiveAdHeight(height) {
+  user().assert(
+      height === WHITELISTED_HEIGHT,
+      'Specified height ' + height +
+          ' in <amp-ad> tag is not one of the approved heights for AdSense.');
+}
+
+/**
+ * Validate <amp-ad> data and validate height for responsive ad units.
+ */
+export function validateAdSenseData(data) {
+  // TODO: check mandatory fields
+  validateData(data, [],
+      ['adClient', 'adSlot', 'adHost', 'adtest', 'tagOrigin', 'experimentId',
+        'ampSlotIndex', 'adChannel', 'autoFormat']);
+  if (data.autoFormat === 'rspv') {
+    validateResponsiveAdHeight(data.height);
```

Comment:
I'm not 100% sure how adsense(global, data) gets called, but I strongly suspect that 'data' there is computed from element.dataset like in amphtml/src/3p-frame.js, so it probably won't have 'height' and 'width' in the general case.